src/redux-middleware/pullBeforeMove.ts

```diff
@@ -0,0 +1,35 @@
+import { ThunkMiddleware } from 'redux-thunk'
+import pull from '../action-creators/pull'
+import { isPending, pathExists, simplifyPath } from '../selectors'
+import { hashContext, head, parentOf, pathToContext } from '../util'
+import { State } from '../util/initialState'
+
+/**
+ * Middleware that fetches any pending descendants before a thought-move operation.
+ */
+const pullBeforeMove: ThunkMiddleware<State> = ({ getState, dispatch }) => {
+
+  return next => async action => {
+
+    if (action.type === 'existingThoughtMove') {
+      const state = getState()
+
+      const oldContext = pathToContext(simplifyPath(state, action.oldPath))
+
+      const newContext = pathToContext(simplifyPath(state, action.newPath))
+
+      const updatedNewContext = [...parentOf(newContext), head(oldContext)]
+      const conflictedPath = pathExists(state, updatedNewContext)
```

Comment:
This check is a bit too narrow. The destination could be pending, but also a descendant of the destination could be pending which could also be a conflict. I have added a failing unit test in e40d11f90 for clarity.

Reply:
Thanks, I'll address this